testomania/src/main/java/com/earth/testomania/technical/data/repository/QuizRepositoryImpl.kt

```diff
@@ -15,14 +19,119 @@ class QuizRepositoryImpl @Inject constructor(
     private val quizApi: QuizApi
 ) : QuizRepository {
 
-    override suspend fun getQuizList(params: QuizCategory): Flow<DataState<List<TechQuizDTO>>> =
+    override suspend fun getQuizList(params: QuizCategory): Flow<DataState<List<Quiz>>> =
         flow {
             quizApi.getQuizList(
                 category = params.category,
                 tags = params.tag
             ).apply {
-                if (isSuccessful) emit(DataState.Success(SuccessMetaData(), payload = body()))
-                else emit(DataState.Error(ErrorMetaData(null)))
+                if (isSuccessful) {
+                    try {
+                        val quizList = mutableListOf<Quiz?>()
+                        body()?.forEach {
+                            quizList += convertTechQuizDTOtoDataObject(it)
+                        }
+                        emit(
+                            DataState.Success(
+                                SuccessMetaData(),
+                                payload = quizList.filterNotNull()
+                            )
+                        )
+                    } catch (e: Exception) {
+                        emit(DataState.Error(ErrorMetaData(e)))
+                    }
+                } else emit(DataState.Error(ErrorMetaData(null)))
             }
         }
+
+
+    private fun convertTechQuizDTOtoDataObject(techQuizDTO: TechQuizDTO) = with(techQuizDTO) {
```

Comment:
I can't understand, this method/model is generic in scope of this concrete category or in whole application

Reply:
Maybe the name is a little confusing. This function converts DTO to Quiz (com.earth.testomania.common.model.Quiz) which is (should be) generic in the whole project. 
So Idea is that any DTO which will come from different repositories will be converted to Quiz  Model and Ui and Ui ViewModel knows how to work with this Quiz model, so integrating new DTO's from other API should be easy, cause what we need is just to create another repository, retrieve according DTO from this repo and convert to Quiz model and it should work.
This is idea of generalizing Quiz Model
@shalva97  Please add your thoughts as well